compiler/rustc_mir_transform/src/inline.rs

```diff
@@ -424,13 +424,6 @@ impl<'tcx> Inliner<'tcx> {
         debug!("    final inline threshold = {}", threshold);
 
         // FIXME: Give a bonus to functions with only a single caller
-        let diverges = matches!(
-            callee_body.basic_blocks[START_BLOCK].terminator().kind,
-            TerminatorKind::Unreachable | TerminatorKind::Call { target: None, .. }
-        );
-        if diverges && !matches!(callee_attrs.inline, InlineAttr::Always) {
-            return Err("callee diverges unconditionally");
-        }
```

Comment:
There is no mir-opt test change in this PR.
Could you add the contents of `tests/codegen/unchecked_shifts.rs` as mir-opt tests? The `Inline.diff` should be enough.
Perhaps the `PreCodegen` so we can see what changes before/after the inliner patch.

Reply:
I find the changes in MIR diffs exceptionally hard to understand, but I'm happy to add that in if it helps other people understand what's going on. I mostly just look at the `PreCodegen` MIR and `git diff` that between compilers.